### PR TITLE
Bump to condalock 1.0, Add pip packages 

### PR DESCRIPTION
**Pull request by scottyhq**
* adds more packages for dataviz tutorial
* bumps to conda-lock 1.0 which handles both conda and pip
   * modify github action to use environment-lock.yml which contains both conda and pip packages

addresses https://github.com/uwhackweek/jupyterbook-template/issues/88, https://github.com/uwhackweek/jupyterbook-template/issues/93, https://github.com/uwhackweek/jupyterbook-template/issues/61

### CODE REVIEW

File: binder/environment.yml [hunk 1]
```diff
@@ -515,3 +519,6 @@ dependencies:
   - sphinxcontrib-serializinghtml=1.1.5=pyhd8ed1ab_1
   - myst-nb=0.13.1=pyhd8ed1ab_0
   - jupyter-book=0.12.2=pyhd8ed1ab_0
+  - pip:
+    - jinja-markdown === 1.210911 --hash=md5:None
+    - rio-stac === 0.3.2 --hash=md5:None
```
Comment (scottyhq) [inline binder/environment.yml:524]: not sure where --hash=md5:None is coming from and is causing a problem

Comment (scottyhq) [reply binder/environment.yml:524]: https://github.com/conda-incubator/conda-lock/issues/172

**Comment by scottyhq** [comment]
not sure why CI takes 5min on linux but 11min on macos... hopefully speed will improve with environment caching after this is merged

File: binder/environment.yml [hunk 2]
```diff
@@ -203,7 +203,7 @@ dependencies:
   - pickleshare=0.7.5=py_1003
   - postgresql=14.2=h2510834_0
   - poyo=0.5.0=py_0
-  - proj=8.2.1=h277dcde_0
+  - proj=9.0.0=h93bde94_1
```
Comment (scottyhq) [inline binder/environment.yml:206]: might want to pin this at 8

File: binder/environment.yml [hunk 3]
```diff
@@ -413,20 +415,20 @@ dependencies:
   - scikit-learn=1.0.2=py39h4dfa638_0
   - tifffile=2022.2.9=pyhd8ed1ab_0
   - wcwidth=0.2.5=pyh9f0ad1d_2
-  - xarray=2022.3.0=pyhd8ed1ab_0
+  - xarray=0.21.1=pyhd8ed1ab_0
```
Comment (scottyhq) [inline binder/environment.yml:418]: https://github.com/gjoseph92/stackstac/issues/136

File: binder/environment.yml (hunk 2 above)
Comment (jomey) [reply binder/environment.yml:206]: What are the concerns to pin this?
If conda is happy to resolve 9, I would keep it

File: conda/lock-environment.sh [hunk 4]
```diff
@@ -23,14 +23,20 @@ if [[ ! -s "${ENV_FILE}" ]]; then
...
 ## Generate environment.yml for binder compatibility
 printf "Generate environment.yml for BinderHub \n"
-conda-lock lock --mamba -f ${ENV_FILE} -p linux-64 -k env
-mv conda-linux-64.lock.yml ../binder/environment.yml
+conda-lock render -k env
+
+# Temporary fix for https://github.com/conda-incubator/conda-lock/issues/172
+sed -i.bak -r 's/--hash=md5:None//' conda-linux-64.lock.yml 
+sed -i.bak -r 's/--hash=md5:None//' conda-osx-64.lock.yml
+rm *.yml.bak
```
Comment (jomey) [inline conda/lock-environment.sh:37]: Might want to log this as an issue.
It is easy to forget temporary fixes in my experience

**Review by jomey — APPROVED** [review]

**Comment by jomey** [comment]
Will we transfer this also to the HW service repo template?

File: conda/lock-environment.sh (hunk 4 above)
Comment (scottyhq) [reply conda/lock-environment.sh:37]: just in the past major version bumps in proj have caused some confusing with ordering of lat/lon in gdal/rasterio... flagging to just keep an eye on it. I'll open an issue!

**Comment by scottyhq** [comment]
> Will we transfer this also to the HW service repo template? 

Yes. I've linked to the relevant issues in the template repository in the top comment

**Pull request merged**